### PR TITLE
Adapter final polish

### PR DESCRIPTION
**Pull request by bretambrose**
Adds some more logging to the adapter implementation

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by sbSteveK — APPROVED** [review]
Looks good. The focus on logging in this PR was the impetus for wondering about adding an mqtt adapter specific log subject. A lot of these logs feel on the fence and could fall on either side of making more sense as a LS mqtt client or mqtt adapter. It'd be fine to just keep it as is. Your call!

**Review by xiazhvera — APPROVED** [review]
Trivial fixes. LGTM.

File: source/v5/mqtt3_to_mqtt5_adapter.c [hunk 1]
```diff
@@ -2270,6 +2425,15 @@ static uint16_t s_aws_mqtt_client_connection_5_subscribe(
 
 error:
 
+    ;
```
[the remark this answers is not in the document]
Comment (bretambrose) [reply source/v5/mqtt3_to_mqtt5_adapter.c:2428]: Those are necessary.  Take one out and see what happens.

[the remark this answers is not in the document]
Comment (bretambrose) [reply source/v5/mqtt3_to_mqtt5_adapter.c:2428]: Less enigmatically, it's just not legal C99 to introduce a variable scope-wise directly after a label.  I don't know the actual rule or correct terminology though.

**Pull request closed without merging**